enhancements/upgrade-bundle.md

```diff
@@ -0,0 +1,138 @@
+
+## Motivation
+
+Testing operator upgrades remains difficult today due to the number of steps to prepare for an upgrade and the lack of tooling that automates those steps. Having high quality Operators is crucial to the success of the Operator Framework and thereby operator developers should have a tool to use to perform an operator upgrade.
+
+### Goals
+
+Provide a command line that upgrades an operator version using OLM.
+
+### Non-Goals
+
+## Proposal
+
+The Operator SDK currently supports running an operator with OLM using the
+`run` command with the `--olm` flag. 
+This enhancement will add a new subcommand, `bundle-upgrade`,
+to the `run` command. This new command, `run bundle-upgrade`, will perform the upgrade of an operator from a previous version to a new specified version.
+
+Currently, the run command has different subcommands: 
+
+ * `run local`
+ * `run packagemanifests`
+ * `run bundle` - currently being implemented, design document is covered by [run bundle enhancement](https://github.com/operator-framework/enhancements/blob/master/enhancements/run-bundle.md)
+
+#### Command Line Interface
+The new command line will look as follows:
+
+```
+operator-sdk run bundle-upgrade (alias: bu) <bundle-image> --from-index=<> --starting-csv=<> [--channel=<>] [--operator-namespace=<>] [--install-mode=<>]
+```
+
+* Positional arguments:
+    * `bundle-image`
+* Required flags:
+    * `--index-image` 
+    * `--starting-csv`
+* Optional flags:
+    * `--channel` 
+    * `--operator-namespace`
+    * `--install-mode`
+
+* `<bundle-image>` is a positional argument that specifies the bundle image that we want to upgrade to
+* `--index-image=<>` is a required flag, which contains the operator we’re upgrading from
+* `--starting-csv=<>` is a required flag, which specifies the first version of the CSV that OLM will install as part of the subscription for a given package. 
+Here's an [sample CSV](https://github.com/operator-framework/operator-sdk-samples/blob/master/go/memcached-operator/deploy/olm-catalog/memcached-operator/manifests/memcached-operator.clusterserviceversion.yaml) in which the value of startingCSV is the name i.e. `memcached-operator.v0.17.0`
+* `[--channel=]` is an optional flag that specifies the channel to upgrade from within a subscription (default channel used if subscription does not contain a channel)
+* `[--operator-namespace=]` is an optional flag that specifies the namespace to install the operator. It will default to the `KUBECONFIG` context. (see run bundle)
+* `[--install-mode=]` is an optional flag that specifies the
+  [OLM install mode](https://red.ht/2YK0ksD). (see run bundle)
+  * We support 3 modes:
+    * AllNamespaces
+    * OwnNamespace
+    * SingleNamespace=ns1
+      * NOTE: if ns1 is the same as the operator's namespace, a warning will be
+        printed
+  * When `--install-mode` is not supplied, the default precedence order is:
+    * `OperatorGroup`, if it already exists
```

Comment:
This should be more like:
```markdown
    * If an `OperatorGroup` already exists with the correct namespacing, use it.
    * Else, create an `OperatorGroup` with the following preferences:
    	* If `AllNamespaces` is supported, use this mode.
    	* Else if `OwnNamespace` is supported, use this mode.
    	* Else:
			* If `SingleNamespace` is not supported, return an error saying "no valid install mode is supported"
			* Else return an error saying "--install-mode=SingleNamespace=<namespace> is required"
```

Also move this to an implementation details section.

Reply:
Modified this as per your suggestion. Will move this to implementation details as well